File: SimulatedOS/Process.cpp
#include "Process.h"


Process::Process() {
	pid = 0;
}
Process::Process(std::string n, std::queue<std::string> i, std::queue<int> r, int id)
{
	this->name = n;
	this->instructions = i;
	this->runtime = r;
	this->pid = id;
	this->state = "NEW";
}



std::string Process::getname(void)
{
	return name;
}

std::string Process::getNextInstruction(void)
{
	std::string nextInstruction;
	if (!(instructions.empty())) {
		nextInstruction = instructions.front();
		instructions.pop();
	}
	else {
		nextInstruction = "EXE";
	}
	return nextInstruction;
}

std::queue<std::string> Process::getInstructions(void)
{
	return this->instructions;
}

int Process::getRuntime(void) {
	int r = runtime.front();
	runtime.pop();
	return r;
}

std::string Process::getState(void)
{
	return this->state;
}

int Process::getId(void)
{
	return pid;
}

void Process::setState(std::string s)
{
	this->state = s;
	return;
}

//MAYBE
//void run(CPU c, this*) { c.add(this.nextInstruction(), c.decrementTimer  }

File: SimulatedOS/src/edu/vcu/patelas23/Process.java
package edu.vcu.patelas23;

public class Process {

    static class pcb {
        int pc = 0;
        int memory = 0;
        String name;
        String[] instructions;
        int[] runtimes;
        String state = "NEW";
        public pcb() {

        }
    }

    private String name;
    private String[] instructions;
    String state;
    private int[] runtimes;
    private int id;
    private int memory;

    public Process(String s, String[] i, int[] r, int id, int mem) {
        name = s;
        instructions = i;
        runtimes = r;
        memory = mem;
        this.id = id;

    }

    public String getName() {
        return name;
    }
    public int getId() {
        return id;
    }

    public void addMaskableInterrupt() {

    }
    public void addUnmaskableInterrupt() {

    }

}
